storage/src/main/java/io/delta/storage/internal/S3LogStoreUtil.java

```diff
@@ -66,7 +66,8 @@ private static RemoteIterator<S3AFileStatus> s3ListFrom(
                         new ListObjectsV2Request()
                                 .withBucketName(s3afs.getBucket())
                                 .withMaxKeys(maxKeys)
-                                .withPrefix(s3afs.pathToKey(parentPath))
+                                .withPrefix(s3afs.pathToKey(parentPath) + "/")
```

Comment:
how do you know that `s3afs.pathToKey(parentPath)` won't already end with a `/`?

Reply:
https://github.com/delta-io/delta/blob/6dacf13e964562098bb4d986979c126ec1f01947/storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java#L200
In `org.apache.hadoop.fs.Path.getParent()`，
```
if (lastSlash==-1) {
      parent = CUR_DIR;
    } else {
      parent = path.substring(0, lastSlash==start?start+1:lastSlash);
    }
```